modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewKeyInputTest.java

```diff
@@ -1109,17 +1131,28 @@ private boolean isAnchor(int row, int col) {
         sm.setSelectionMode(SelectionMode.MULTIPLE);
         sm.setCellSelectionEnabled(true);
         sm.select(0, col0);
-        keyboard.doRightArrowPress(KeyModifier.getShortcutKey());
-        keyboard.doRightArrowPress(KeyModifier.getShortcutKey());
+
+        if (orientation == NodeOrientation.LEFT_TO_RIGHT) {
+            keyboard.doRightArrowPress(KeyModifier.getShortcutKey());
+            keyboard.doRightArrowPress(KeyModifier.getShortcutKey());
+        } else if (orientation == NodeOrientation.RIGHT_TO_LEFT) {
+            keyboard.doLeftArrowPress(KeyModifier.getShortcutKey());
+            keyboard.doLeftArrowPress(KeyModifier.getShortcutKey());
+        }
```

Comment:
having such if blocks in all tests regarding horizontal navigation feels like the parametrization isn't quite complete, IMO: the test method shouldn't need to worry, instead just call semantic horizontal navigation methods.

A simple change would be to extract that differentiation into a dedicated method, like f.i. forward/backward, test would get simpler (and make it easier to add those that are missing :)

    @Test
    public void testForwardFocus() {
        sm.setCellSelectionEnabled(true);
        sm.select(0, col0);
        // current
        //if (orientation == NodeOrientation.LEFT_TO_RIGHT) {
        //    keyboard.doRightArrowPress(KeyModifier.getShortcutKey());
        //} else {
        //    keyboard.doLeftArrowPress(KeyModifier.getShortcutKey());
        //}
        // extracted
        forward(KeyModifier.getShortcutKey());
        assertTrue("selected cell must still be selected", sm.isSelected(0, col0));
        assertFalse("next cell must not be selected", sm.isSelected(0, col1));
        TablePosition focusedCell = fm.getFocusedCell();
        assertEquals("focused cell must moved to next", col1, focusedCell.getTableColumn());
    }
    
    /**
     * Orientation-aware horizontal navigation with arrow keys.
     * @param modifiers the modifiers to use on keyboard
     */
    protected void forward(KeyModifier... modifiers) {
        if (orientation == NodeOrientation.LEFT_TO_RIGHT) {
            keyboard.doRightArrowPress(modifiers);
        } else {
            keyboard.doLeftArrowPress(modifiers);
        }
    }

Reply:
Also, I'm a bit weary about the "else if" (vs a simple "else") - wouldn't it be some kind of setup error if the node orientation is neither rtl nor ltr? If so, I would add a test to check for it once.

Reply:
Initially I had thought about adding separate test file for RTL - something like RTLTableViewKeyInputTest - but, realized that although it's a cleaner approach, we would simply duplicate the tests. Also, the fact is only LEFT/RIGHT key navigation is sensitive to NodeOrientation - hence only a subset of tests needed modification. This is the reason I have parameterized the test.

To your specific question, since it is a parameterized test, only possible values are LTR and RTL which are specified as @Parameterized.Parameters. I don't think, we need additional check for some other value.

Reply:
Regarding your suggestion of having forward/backward semantic methods and calling them in tests - 
There are two types of tests -
1. Key action remains same - asserts differ based on NodeOrientation
2. Key action differs based on NodeOrientation - but, asserts remain same

Example of 1 is - test_rt18488_selectToLeft
Example of 2 is - test_rt18591_cell_1

Your suggestion can be applied only to the tests of type-2. I will try to update tests of this type.

Reply:
ahh .. okay, missed that difference, thanks :) 

Hmm, still feels smelly to have to use the parameter for conditional branches (either on the assumption or on the navigation). Will check later

Reply:
and yes, agree with using parameterized tests at all ... just have this nagging feeling that it's not quite done to the end :)